proposals/src/ledger-replication-to-implement.md

```diff
@@ -37,3 +37,84 @@ transacation proves the validator incorrectly validated a fake storage proof.
 The replicator is rewarded and the validator's staking balance is slashed or
 frozen.
 
+### Storage proof contract logic
+
+Each replicator and validator will have their own storage account. The validator's
+account would be separate from their gossip id similiar to their vote account.
+These should be implemented as two programs one which handles the validator as the keysigner
+and one for the replicator. In that way when the programs reference other accounts, they
+can check the program id to ensure it is a validator or replicator account they are
+referencing.
+
+#### SubmitMiningProof
+```compile_fail
+SubmitMiningProof {
+    slot: u64,
+    sha_state: Hash,
+    signature: Signature,
+};
+keys = [replicator_keypair]
+```
+Replicators create these after mining their stored ledger data for a certain hash value.
+The slot is the end slot of the segment of ledger they are storing, the sha\_state
+the result of the replicator using the hash function to sample their encrypted ledger segment.
+The signature is the signature that was created when they signed a PoH value for the
+current storage epoch. The list of proofs from the current storage epoch should be saved
+in the account state, and then transfered to a list of proofs for the previous epcoh when
+the epoch passes. In a given storage epoch a given replicator should only submit proofs
+for one segment. The program should check that the slot and signature is valid and then store
```

Comment:
The slot is valid as long as it belongs to the current storage epoch? How is the signature verified, the program will need to know what storage_blockhash was used. Is this related to the storage_epoch we're in ? sorry I know some of these might be really basic questions

Reply:
I added a line about that, we can keep a set of valid slots.